Buildings/Fluid/BaseClasses/FlowModels/basicFlowFunction_dp_m_flow.mo

```diff
@@ -0,0 +1,43 @@
+within Buildings.Fluid.BaseClasses.FlowModels;
+function basicFlowFunction_dp_m_flow
+  "Inverse of flow function that computes the flow coefficient"
+  extends Modelica.Icons.Function;
+
+  input Modelica.SIunits.MassFlowRate m_flow
+    "Mass flow rate in design flow direction";
+  input Modelica.SIunits.PressureDifference dp
+    "Pressure difference between port_a and port_b (= port_a.p - port_b.p)";
+  input Modelica.SIunits.MassFlowRate m_flow_small
+    "Minimum value of mass flow rate guarding against k=(0)/sqrt(dp)";
+  input Modelica.SIunits.PressureDifference dp_small
+    "Minimum value of pressure drop guarding against k=m_flow/(0)";
+  output Real k
+    "Flow coefficient";
+protected
+  Modelica.SIunits.PressureDifference dpSqrt
+    "Regularized square root value of pressure drop";
+  Modelica.SIunits.MassFlowRate mPos_flow
+    "Regularized absolute value of mass flow rate";
+algorithm
+  dpSqrt := Buildings.Utilities.Math.Functions.regNonZeroPower(
+    dp, 0.5, dp_small);
+  mPos_flow := Buildings.Utilities.Math.Functions.regNonZeroPower(
+    m_flow, 1, m_flow_small);
```

Comment:
Is there a particular reason for using `regNonZeroPower` with exponent `1` rather than using `smoothMax(m_flow, -m_flow, m_flow_small)`? If not, then `smoothMax` would be preferred as it is more efficient.

Reply:
@mwetter `regNonZeroPower` is used originally to guard against a zero value since we need to take the inverse of the flow coefficient when calling
```
Buildings.Fluid.Actuators.BaseClasses.exponentialDamper_inv(
      kThetaSqRt=sqrt(2*rho)*A/kDam, kSupSpl=kSupSpl, ySupSpl=ySupSpl, invSplDer=invSplDer)
```
However, thinking again about it, it would be more efficient to use the actual value of the loss coefficient `kTheta` in the above equation and to compute directly the square inverse of the flow coefficient in `basicFlowFunction_dp_m_flow` as

```
  output Real kSquInv
    "Square inverse of flow coefficient";
protected 
  Modelica.SIunits.PressureDifference dpPos=
    Buildings.Utilities.Math.Functions.smoothMax(dp, -dp, dp_small)
    "Regularized absolute value of pressure drop";
  Real mSqu_flow = Buildings.Utilities.Math.Functions.smoothMax(
    m_flow^2, m_flow_small^2, m_flow_small^2)
    "Regularized non zero square value of mass flow rate";
algorithm 
  kSquInv := dpPos / mSqu_flow;
```

The drawback is that the input variables of `exponentialDamper_inv` and `basicFlowFunction_dp_m_flow` are no more consistent with the variables of the original functions `exponentialDamper` and `basicFlowFunction_dp`. But I think this is acceptable as the inverse functions are unlikely to be used outside of the pressure independent model.

Please tell me if you think the changes in commit [5837774](https://github.com/lbl-srg/modelica-buildings/pull/1918/commits/58377741dc08e3d9b072c055a8bdfc3670981914) are then worth considering.

Reply:
I think this is fine; but if they are no longer consistent and only used in that model, you could put the function directly in `protected` section of the model. Then no other model can use it, and consistency is less important.

Reply:
Here is the updated performance comparison with Dymola
![Screen Shot 2020-05-04 at 3 14 56 PM](https://user-images.githubusercontent.com/10965994/81018929-5be13080-8e1a-11ea-8321-233f220ea3ca.png)
and Optimica
![Screen Shot 2020-05-04 at 3 14 41 PM](https://user-images.githubusercontent.com/10965994/81018938-66032f00-8e1a-11ea-8ed3-018dfcf8c72b.png)
So no noticeable impact.